src/devices/bus/pce_ctrl/pachinko.h

```diff
@@ -0,0 +1,62 @@
+// license:BSD-3-Clause
+// copyright-holders:kmg
+/**********************************************************************
+
+    NEC PC Engine Coconuts Japan CJPC-101 Pachinko Controller
+
+**********************************************************************/
+
+#ifndef MAME_BUS_PCE_CTRL_PACHINKO_H
+#define MAME_BUS_PCE_CTRL_PACHINKO_H
+
+#pragma once
+
+
+#include "pcectrl.h"
+
+
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> pce_pachinko_device
+
+class pce_pachinko_device : public device_t,
+							public device_pce_control_port_interface
+{
+public:
+	// construction/destruction
+	pce_pachinko_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	// optional information overrides
+	virtual ioport_constructor device_input_ports() const override;
+
```

Comment:
This should be `protected`.  It’s declared before `device_start` in the `device_t` declaration.